### PR TITLE
ci: fix reload config for empty pinned image test

### LINKED ISSUE #8324: Flaky test on Kata runtime (reload_config)
### Which jobs are failing?

fedora-kata - reload_config

### Which tests are failing?

`kata_skip_reload_config `
https://github.com/cri-o/cri-o/blob/88fe8766ede790f465e1e86b0eccba0818736b32/contrib/test/ci/vars.yml#L109-L110

### Since when has it been failing?

This issue is used to track the status of flaky tests. The test is disabled after merge:

https://github.com/cri-o/cri-o/pull/8213 https://github.com/cri-o/cri-o/pull/7976

### Testgrid link

_No response_

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by littlejawa**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:
Modify the "wait_for_log" function in helpers.bash to allow it to catch multiple occurrences of the same message.
Use that in the failing test to synchronize and wait for config reload to be finished before checking the content of the configuration.

#### Which issue(s) this PR fixes:
Fixes #8324

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by littlejawa** [comment]
@sohankunkerkar, @kwilczynski - as you reviewed the original PR, could you get a look at this one?

**Comment by haircommander** [comment]
/approve
/retest

@cri-o/cri-o-maintainers PTAL

**Comment by littlejawa** [comment]
@cri-o/cri-o-maintainers - can we merge this one ?

**Comment by sohankunkerkar** [comment]
@littlejawa could you address https://github.com/cri-o/cri-o/actions/runs/10469089567/job/28991329813?pr=8399?

**Comment by sohankunkerkar** [comment]
/hold
Feel free to unhold when the test is fixed.

File: test/helpers.bash
Comment (kwilczynski) [inline test/helpers.bash:525]: Since the following, this regular expression might not match any more:

- https://github.com/cri-o/cri-o/pull/8592

**Comment by sohankunkerkar** [comment]
@littlejawa, it looks like CI is almost green! Could you squash the last four commits and prepare this PR for merge?

**Comment by littlejawa** [comment]
> @littlejawa, it looks like CI is almost green! Could you squash the last four commits and prepare this PR for merge?

Done. Thanks !

**Review by sohankunkerkar — APPROVED** [review]
/lgtm

**Comment by kwilczynski** [comment]
OK. Will cherry-pick manually.

**Pull request merged**
